sql/core/src/test/scala/org/apache/spark/sql/scripting/SqlScriptingExecutionNodeSuite.scala

```diff
@@ -40,10 +41,38 @@ class SqlScriptingExecutionNodeSuite extends SparkFunSuite with SharedSparkSessi
       Origin(startIndex = Some(0), stopIndex = Some(description.length)),
       isInternal = false)
 
+  case class TestWhileCondition(description: String)
+    extends SingleStatementExec(
```

Comment:
we are not using anything from `SingleStatementExec` - you can make it extend `LeafStatementExec` only, to simplify things.

Reply:
Added DummyLogicalPlan to TestWhileCondition.